Gems/Atom/RHI/DX12/Code/Source/RHI/CommandList.cpp

```diff
@@ -707,6 +750,31 @@ namespace AZ
             {
                 GetCommandList()->OMSetRenderTargets(renderTargetCount, colorDescriptors, false, nullptr);
             }
+
+#ifdef AZ_DX12_VRS_SUPPORT
+            if (RHI::CheckBitsAll(GetDevice().GetFeatures().m_shadingRateTypeMask, RHI::ShadingRateTypeFlags::PerImage))
+            {
+                auto commandList5 = DX12ResourceCast<ID3D12GraphicsCommandList5>(GetCommandList());
+                AZ_Assert(commandList5, "Failed to cast command list to ID3D12GraphicsCommandList5");
+                if (commandList5)
+                {
+                    if (shadingRateAttachment)
```

Comment:
I suggest checking for a shading rate attachment first, before doing the COM interface query and asserts and capability checking (since in most cases here, the SR attachment won't be active)